src/gui/editorwidgets/core/qgseditorwidgetregistry.h

```diff
@@ -68,6 +97,16 @@ class GUI_EXPORT QgsEditorWidgetRegistry : public QObject
     ~QgsEditorWidgetRegistry();
 
     /**
+     * Find the best editor widget and its configuration for a given field.
+     *
+     * @param vl        The vector layer for which this widget will be created
+     * @param fieldName The field name on the specified layer for which this widget will be created
+     *
+     * @return The id of the widget type to use and its config
+     */
+    QgsWidgetSetup findBest( const QgsVectorLayer& vl, const QString& fieldName ) const;
```

Comment:
That is a weird convention. To me, passing by pointer means giving out ownership. But OK, if it's the convention...